docs/developers/developing_iree/profiling_with_tracy.md

```diff
@@ -27,6 +28,22 @@ a PDF manual that's part of each numbered release.
 or
 [view in browser](https://docs.google.com/viewer?url=https://github.com/wolfpld/tracy/releases/latest/download/tracy.pdf).
 
+## Overview
+
+We will go through each steps below, but here is an overview. It highlights the simpler subset of instructions when only instrumentation is needed, vs. the additional steps needed when Sampling is also wanted.
+
+Component | Instrumentation only | Instrumentation and Sampling
+--- | --- | ---
+Build Tracy capture (`iree-tracy-capture`) | Base instructions below for [dependencies](#install-dependencies) and [build](#build-the-tracy-tools) | Same
+Build Tracy profiler (`iree-tracy-profiler`) | Base instructions below for [dependencies](#install-dependencies) and [build](#build-the-tracy-tools) | Same plus [`capstone-next` instructions](#do-you-need-capstone-next) for CPU disassembly to work
+Build the IREE compiler (`iree-compile`) for profiling your own modules  | [Nothing particular](#build-the-iree-compiler-iree-compile) | Same
+Build the IREE compiler (`iree-compile`) for profiling the compiler itself | [Also need](#build-the-iree-compiler-iree-compile) Cmake setting: `IREE_ENABLE_COMPILER_TRACING` | Same
+Compile your IREE module (run `iree-compile`) | [Nothing particular](#compile-your-iree-module-run-iree-compile) | [Also need](#additional-steps-for-sampling) to pass `--iree-llvm-link-embedded=false`, and also `--iree-llvm-debug-symbols=true` but that is currently default.
```

Comment:
Might be worth noting that the `--iree-llvm` flags only matter if you are using `--iree-hal-target-backends=llvm-cpu`. If you are using `vulkan` or another backend then you can skip them.